cmd/testdata/gatewayd.yaml

```diff
@@ -18,7 +18,7 @@ metrics:
 
 clients:
   default:
-    activeWrites:
+    active-writes:
```

Comment:
yeah right, the idea might be a bit complex. We need to convert all `configuration groups` and` configuration blocks `that are keys in our array system (they are case-sensitive) into lowercase. During validation, we will check if they are in lowercase.

If we want to handle this in the following part of the code: https://github.com/gatewayd-io/gatewayd/blob/main/config%2Fconfig.go#L339
we need to convert `UPPERCASE` to `LowerCamelCase`. If I understand correctly, those packages cannot convert UPPERCASE to LowerCamelCase, or we need to configure the `acronym` beforehand. Please correct me if I am wrong.
https://go.dev/play/p/akzbPSx8Dyq

Reply:
It's indeed more complex than I thought, but I think it's a better way than what you try, especially because your PR would imply another breaking changes for user as the activeWrites would require to update configuration files.

And it would also introduce something a new "case" type in the YAML, where camelCase would be mixed with snake_case

Reply:
> we need to convert UPPERCASE to LowerCamelCase. If I understand correctly, those packages cannot convert UPPERCASE to LowerCamelCase, or we need to configure the acronym 

I would expect using SCREAMING_SNAKE_CASE in env variables

You can switch from screaming snake case to the other

Reply:
> PR would imply another breaking changes for user as the activeWrites would require to update configuration files.

In the PR, we convert before validation, so we don’t have breaking changes; we lowercase everything anyway.
So, it doesn’t matter how users write configuration files; we lowercase them.
However, I agree that we need to have a consistent case type in our YAML.

> I would expect using SCREAMING_SNAKE_CASE in env variables

Currently https://docs.gatewayd.io/using-gatewayd/configuration/#environment-variables we use `_` to separate nested items in YAML. Using SCREAMING_SNAKE_CASE can be more complex and create more edge cases.”

Reply:
I can assume things are more complex than what I could assume by simply reading 2 PRs

I simply came and shared my feedbacks

I'm sure you can find something.

The PR might be merged like that maybe, but I'm annoyed by the current changes of case for activeWrites,(which is also inconsistent, and both agree about it), so maybe you could add quick and dirty hack only for keeping activeWrites with current case in the YAML, merge with the hack and prepare another PR to provide a better solution.

I'm worried about the back and forth changes that users have to do in the YAML configuration file.

A new node, move existing conf to it, rename it to active-writes, then maybe later come back to activeWrites…

Reply:
Thanks a lot for your feedback, it’s super helpful! I’ll get back to ActiveWrites, and we’ll keep the same case type for the YAML config file without any breaking changes.

Reply:
Great then